be/src/io/fs/file_system.h

```diff
@@ -64,6 +65,43 @@ struct FileInfo {
     bool is_file;
 };
 
+struct FsListGenerator {
+    FsListGenerator() = default;
+    virtual ~FsListGenerator() = default;
+
+    virtual Status init() = 0;
```

Comment:
what is init() for? subclass should be initialized when it is created, we dont need to call baseclass.init().